synapse/events/spamcheck.py

```diff
@@ -208,9 +212,21 @@ async def check_registration_for_spam(
             # spam checker
             checker = getattr(spam_checker, "check_registration_for_spam", None)
             if checker:
-                behaviour = await maybe_awaitable(
-                    checker(email_threepid, username, request_info)
-                )
+                checker_args = inspect.getfullargspec(checker)
```

Comment:
I think really you want `inspect.signature` here rather than `getfullargspec`. (see https://docs.python.org/3/library/inspect.html#inspect.getfullargspec)

Reply:
Oh, good to know that `getfullargspec` is on the way out. I suppose we should also update the other users of it in the codebase.